tests/Unit/ControlSystem/Systems/Test_Expansion_System.cpp

```diff
@@ -0,0 +1,171 @@
+  ActionTesting::emplace_array_component<element_component>(
+      make_not_null(&runner), ActionTesting::NodeId{0},
+      ActionTesting::LocalCoreId{0}, 0);
+  ActionTesting::emplace_nodegroup_component<observer_component>(
+      make_not_null(&runner));
+
+  ActionTesting::set_phase(make_not_null(&runner), metavars::Phase::Testing);
+
+  const BinaryTrajectories binary_trajectories{initial_separation};
+
+  const std::string& expansion_name = system_helper.expansion_name();
+
+  // Create coordinate maps for mapping the PN expansion to the "grid" frame
+  // where the control system does its calculations
+  // The outer boundary is at 1000.0 so that we don't have to worry about it.
+  ExpansionMap expansion_map{1000.0, expansion_name,
+                             expansion_name + "OuterBoundary"s};
+
+  CoordMap coord_map{expansion_map};
+
+  // Get the functions of time from the cache to use in the maps
+  const auto& cache = ActionTesting::cache<element_component>(runner, 0);
+  const auto& functions_of_time =
+      Parallel::get<domain::Tags::FunctionsOfTime>(cache);
+
+  const auto position_function = [&binary_trajectories](const double time) {
+    const double separation = binary_trajectories.separation(time);
+    return std::pair<std::array<double, 3>, std::array<double, 3>>{
+        {-0.5 * separation, 0.0, 0.0}, {0.5 * separation, 0.0, 0.0}};
+  };
+
+  // Run the actual control system test.
+  system_helper.run_control_system_test(runner, final_time, make_not_null(&gen),
+                                        position_function, coord_map);
+
+  // Grab results
+  const std::array<double, 3>& grid_position_of_a =
+      system_helper.grid_position_of_a();
+  const std::array<double, 3>& grid_position_of_b =
+      system_helper.grid_position_of_b();
+
+  // Our expected positions are just the initial positions
+  const std::array<double, 3> expected_grid_position_of_a{
+      {-0.5 * initial_separation, 0.0, 0.0}};
+  const std::array<double, 3> expected_grid_position_of_b{
+      {0.5 * initial_separation, 0.0, 0.0}};
+
+  const auto& expansion_f_of_t =
+      dynamic_cast<domain::FunctionsOfTime::PiecewisePolynomial<DerivOrder>&>(
+          *functions_of_time.at(expansion_name));
+
+  const double exp_factor = expansion_f_of_t.func(final_time)[0][0];
+
+  // The control system gets more accurate the longer you run for. However, this
+  // is the floor of our accuracy for a changing function of time.
+  Approx custom_approx = Approx::custom().epsilon(5.0e-6).scale(1.0);
+  const double expected_exp_factor =
+      binary_trajectories.separation(final_time) / initial_separation;
+  CHECK(custom_approx(expected_exp_factor) == exp_factor);
+
+  CHECK_ITERABLE_CUSTOM_APPROX(expected_grid_position_of_a, grid_position_of_a,
+                               custom_approx);
+  CHECK_ITERABLE_CUSTOM_APPROX(expected_grid_position_of_b, grid_position_of_b,
+                               custom_approx);
+}
+
+// [[Timeout, 60]]
```

Comment:
Ouch!  This needs to be sped up or split up.

Reply:
Oops, this was a relic from debugging. It fits within the default test time.